packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -131,18 +137,25 @@ export class WaferMap extends FoundationElement {
             this.dieLabelsSuffix,
             this.maxCharacters
         );
-        this.renderer = new RenderingModule(this.dataManager, this.canvas);
-        this.zoomHandler = new ZoomHandler(
-            this.canvas,
-            this.zoomContainer,
-            this.dataManager,
-            this.renderer,
-            this.canvasSideLength
+
+        const renderer = new RenderingModule(dataManager, this.canvas);
+        this.eventHandler = new EventHandler(
```

Comment:
It seems strange to me to need to destroy and recreate all the event handlers during each render. Once the correct attach and detach behavior is implemented in this PR we should make sure this is actually necessary to do every render.

Reply:
This is done mainly because of the current implementation of the DataManager - which has to be recreated not just when new data is being fed to the component, but also when the size of the wafermap is changing (canvasSideLength). This is happening because of the computations being performed within. Since this module is also used by the rendering module and EventCoordinator, both have to be recreated as well.